slrealizer/utils/utils.py

```diff
@@ -1,23 +1,58 @@
-#==============================================
-
+#import numpy as np
+#from fractions import Fraction
+#import math
 import numpy as np
-import pandas
-import desc.slrealizer
-from fractions import Fraction
-import math
+#import pandas as pd
+
+def e1e2_to_ephi(e1, e2):
+    e = np.power(np.power(e1, 2.0) + np.power(e2, 2.0), 0.5)
+    phi = np.arctan(e2/e1)
+    return e, phi
+
+def ephi_to_e1e2(e, phi):
+    e1 = e*np.cos(phi)
+    e2 = e*np.sin(phi)
+    return e1, e2
+
+def get_1D_columns(multidimColNames, table):
+    totalColDict = {}
+    for mc in multidimColNames:
+        colSize = 4 #table[mc].shape[1] is always 4
+        colNames = [mc + '_' + str(c) for c in range(colSize)]
+        colValues = [table[mc][:, c].data for c in range(colSize)]
+        colDict = dict(zip(colNames, colValues))
+        totalColDict.update(colDict)
+    return totalColDict
+
+def hlr_to_sigma(hlr):
+    return hlr/np.sqrt(2.0*np.log(2.0))
 
-#==============================================
+def fwhm_to_sigma(fwhm):
+    return fwhm/np.sqrt(8.0*np.log(2.0))
 
-"""
-This file contains methods that help generate the toy catalog for the SLRealizer.
-"""
+def pixel_to_physical(pixelPos, canvasSize, pixel_scale):
+    return (pixelPos - 0.5*canvasSize - 0.5)*pixel_scale
 
-def noissify_data(mean, stdev):
+def physical_to_pixel(physicalPos, canvasSize, pixel_scale):
+    return physicalPos/pixel_scale + 0.5*canvasSize + 0.5
+
+def from_flux_to_mag(flux, zeropoint_mag=0.0, from_unit=None, to_unit=None):
+    if from_unit=='nMgy':
+        zeropoint_mag=22.5
+    return zeropoint_mag-2.5*np.log10(flux)
+
+def from_mag_to_flux(mag, zeropoint_mag=0.0, from_unit=None, to_unit=None):
+    if to_unit=='nMgy':
+        zeropoint_mag=22.5
+    return np.power(10.0, -0.4*(mag - zeropoint_mag))
+
+def add_noise(mean, stdev, measurement=1.0):
     """
     Given a mean and a standard deviation of a measurement, add a noise to the data
```

Comment:
Add a comment that the "measurement" kwarg is for computing fractional, not absolute, errors.

Reply:
Done